service/src/test/java/bio/terra/workspace/app/configuration/external/controller/FolderApiControllerTest.java

```diff
@@ -0,0 +1,545 @@
+    String serializedUpdateResponse =
+        updateFolderExpectCode(
+                workspaceId,
+                folderId,
+                newDisplayName,
+                newDescription,
+                parentFolderId,
+                updateParent,
+                HttpStatus.SC_OK)
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    return objectMapper.readValue(serializedUpdateResponse, ApiFolder.class);
+  }
+
+  private ResultActions updateFolderExpectCode(
+      UUID workspaceId,
+      UUID folderId,
+      String newDisplayName,
+      String newDescription,
+      @Nullable UUID parentFolderId,
+      boolean updateParent,
+      int code)
+      throws Exception {
+    return mockMvc
+        .perform(
+            addAuth(
+                patch(String.format(FOLDER_V1_PATH_FORMAT, workspaceId, folderId))
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .characterEncoding("UTF-8")
+                    .content(
+                        getUpdateRequestInJson(
+                            newDisplayName, newDescription, parentFolderId, updateParent)),
+                USER_REQUEST))
+        .andExpect(status().is(code));
+  }
+
+  private String getUpdateRequestInJson(
+      @Nullable String newDisplayName,
+      @Nullable String newDescription,
+      @Nullable UUID parentFolderId,
+      boolean updateParent)
+      throws JsonProcessingException {
+    var requestBody =
+        new ApiUpdateFolderRequestBody()
+            .description(newDescription)
+            .displayName(newDisplayName)
+            .parentFolderId(parentFolderId)
+            .updateParent(updateParent);
+    return objectMapper.writeValueAsString(requestBody);
+  }
+
+  private ApiFolder getFolder(UUID workspaceId, UUID folderId) throws Exception {
+    String folderGetResponse =
+        getFolderExpectCode(workspaceId, folderId, HttpStatus.SC_OK)
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    return objectMapper.readValue(folderGetResponse, ApiFolder.class);
+  }
+
+  private ResultActions getFolderExpectCode(UUID workspaceId, UUID folderId, int code)
+      throws Exception {
+    return mockMvc
+        .perform(
+            addAuth(get(String.format(FOLDER_V1_PATH_FORMAT, workspaceId, folderId)), USER_REQUEST))
+        .andExpect(status().is(code));
+  }
+
+  private ApiFoldersList listFolders(UUID workspaceId) throws Exception {
+    String foldersGetResponse =
+        listFoldersExpectCode(workspaceId, HttpStatus.SC_OK)
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    var retrievedFolders = objectMapper.readValue(foldersGetResponse, ApiFoldersList.class);
+    return retrievedFolders;
+  }
+
+  private ResultActions listFoldersExpectCode(UUID workspaceId, int code) throws Exception {
```

Comment:
Either return void, or convert ResultActions to ApiFolderList. Same for rest of this file.

Reply:
it is used by listFolder which take in the ResultAction and convert it to an ApiFolderList.

When the code is not 200, there's no apiFolderList to convert to. And I want to re-use this code for the `listFolder` helper method.

Reply:
Got it. Do you mind adding this comment

// Can't return ApiFolderList because when code is not 200, there is not ApiFolderList to convert to. Returns ResultActions because this is called by createFolder().